libs/vkd3d/command.c

```diff
@@ -8802,6 +8810,68 @@ static struct d3d12_command_list *unsafe_impl_from_ID3D12CommandList(ID3D12Comma
     return CONTAINING_RECORD(iface, struct d3d12_command_list, ID3D12GraphicsCommandList_iface);
 }
 
+static inline struct d3d12_command_list *d3d12_command_list_from_ID3D12GraphicsCommandListExt(ID3D12GraphicsCommandListExt *iface)
+{
+    struct d3d12_command_list_ext* list_ext = CONTAINING_RECORD(iface, struct d3d12_command_list_ext, ID3D12GraphicsCommandListExt_iface);
+    return CONTAINING_RECORD(list_ext, struct d3d12_command_list, command_list_ext);
+}
+
+static ULONG STDMETHODCALLTYPE d3d12_command_list_ext_AddRef(ID3D12GraphicsCommandListExt *iface)
+{
+    struct d3d12_command_list *command_list = d3d12_command_list_from_ID3D12GraphicsCommandListExt(iface);
+    return ID3D12GraphicsCommandList_AddRef(&command_list->ID3D12GraphicsCommandList_iface);
+}
+
+static ULONG STDMETHODCALLTYPE d3d12_command_list_ext_Release(ID3D12GraphicsCommandListExt *iface)
+{
+    struct d3d12_command_list *command_list = d3d12_command_list_from_ID3D12GraphicsCommandListExt(iface);
+    return ID3D12GraphicsCommandList_Release(&command_list->ID3D12GraphicsCommandList_iface);
+}
+
+static HRESULT STDMETHODCALLTYPE d3d12_command_list_ext_QueryInterface(ID3D12GraphicsCommandListExt *iface,
+        REFIID iid, void **out)
+{
+    TRACE("iface %p, iid %s, out %p.\n", iface, debugstr_guid(iid), out);
+
+    if (IsEqualGUID(iid, &IID_ID3D12GraphicsCommandListExt) ||
+        IsEqualGUID(iid, &IID_IUnknown))
+    {
+        d3d12_command_list_ext_AddRef(iface);
```

Comment:
Not sure how COM rules work here, but IIRC we have to be able to query ID3D12GraphicsCommandList interfaces here.

Reply:
Yes, we need to return the parent here too under its guids.

Reply:
Yeah, the proper way to do this is to implement all COM queries in one place (`d3d12_command_list`) and have this function call that instead.